File: 3365. Rearrange K Substrings to Form Target String.cpp
class Solution {
public:
    bool isPossibleToRearrange(string s, string t, int k) {
        unordered_map<string, int> f;
        int l = s.size() / k;
        for (int i=0; i<s.size(); i+=l) f[s.substr(i, l)]++;
        for (int i=0; i<t.size(); i+=l) {
            auto it = f.find(t.substr(i, l));
            if (it == f.end() || it->second == 0)
                return false;
            it->second--;
        }
        return true;
    }
};

File: 3365. Rearrange K Substrings to Form Target String.py
class Solution:
    def isPossibleToRearrange(self, s: str, t: str, k: int) -> bool:
        f = defaultdict(int)
        l = len(s) // k
        for i in range(0, len(s), l):
            f[s[i:i+l]] += 1
        for i in range(0, len(t), l):
            ss = t[i:i+l]
            if ss not in f or not f[ss]:
                return False
            f[ss] -= 1
        return True
